app/src/main/index.js

```diff
@@ -0,0 +1,73 @@
+'use strict'
+
+import { app, BrowserWindow } from 'electron'
+import * as path from 'path'
+import { format as formatUrl } from 'url'
+
+const is_development = process.env.NODE_ENV !== 'production'
+
+let main_window_keep_alive
+
+function create_main_window() {
+    const window = new BrowserWindow(
+        {
+            width          : 1024,
+            height         : 768,
+            frame          : false,
+        }
+    )
+
+    var search = "";
+    if (app.commandLine.hasSwitch("debug")) {
+        var value = app.commandLine.getSwitchValue("debug");
+        search = "?debug";
+        if (value !== undefined) {
+            search += "=" + value
+        }
+    }
+
+    if (is_development) {
+        window.webContents.openDevTools();
+        var url = `http://localhost:${process.env.ELECTRON_WEBPACK_WDS_PORT}/main/` + search;
+        window.loadURL(url)
+    } else {
+        window.loadURL(formatUrl({
+            pathname : path.join(__dirname.replace(/app\.asar$/, 'static'), '/main/index.html'),
+            protocol : 'file',
+            slashes  : true
+        }) + search)
```

Comment:
This is a really complex expression. Please refactor it to variables. It would be so nicer to write `let url = formatUrl({pathname,protocol,slashes})` instead :)